OpenMTR/Code/Digital/American.cs

```diff
@@ -0,0 +1,108 @@
+                DebugUtils.Log(ex.Message);
+            }
+        }
+
+        private static bool ExtractionFirstPass(Meter meter)
+        {
+            ImageUtils.AdjustImageSkew(meter);
+            meter.ModifiedImage = new Mat(meter.ModifiedImage.Clone(), new Rect(0, 0, meter.ModifiedImage.Width, meter.ModifiedImage.Height / 2));
+            ImageUtils.ColorToGray(meter.ModifiedImage, meter.ModifiedImage);
+            Cv2.GaussianBlur(meter.ModifiedImage, meter.ModifiedImage, new Size(3, 3), 0);
+            Cv2.MorphologyEx(meter.ModifiedImage, meter.ModifiedImage, MorphTypes.Close, ImageUtils.GetKernel(new Size(3, 3)));
+            Cv2.Canny(meter.ModifiedImage, meter.ModifiedImage, 100, 200);
+            List<Rect> readouts = ExtractReadouts(meter);
+            if (readouts.Count == 0)
+            {
+                return false;
+            }
+            List<Rect> extractedDigits = ExtractDigits(meter, readouts);
+            if (extractedDigits.Count != 4)
+            {
+                return false;
+            }
+            //string odometerValue = Odometer.Read(meter, extractedDigits);
+            //DebugUtils.Log($"{meter.FileName}: Read {odometerValue}");
+            //return (odometerValue.Equals(meter.MetaData.MeterRead));
+            return true;
+        }
+
+        private static List<Rect> ExtractReadouts(Meter meter)
+        {
+            Cv2.FindContours(meter.ModifiedImage, out Point[][] contours, out HierarchyIndex[] hierarchy, RetrievalModes.External, ContourApproximationModes.ApproxSimple);
+            List<Rect> readouts = new List<Rect>();
+            foreach (Point[] point in contours)
+            {
+                Rect rect = Cv2.BoundingRect(point);
+                double area = rect.Height * rect.Width; 
+                if (area > 4000 && area < 10000)
+                {
+                    if (rect.Width > (rect.Height * 2) && rect.Width < (rect.Height * 3))
+                    {
+                        readouts.Add(rect);
+                    }
+                }
+            }
+            return readouts;
+        }
+
+        private static List<Rect> ExtractDigits(Meter meter, List<Rect> readouts)
+        {
+            List<Rect> rectangles = new List<Rect>(); 
+            foreach (Rect readout in readouts)
+            {
+                List<Rect> filteredRect = new List<Rect>();
+                Mat roi = new Mat(meter.SourceImage.Clone(), readout);
+                ImageUtils.AdjustImageSkew(roi);
+                ImageUtils.ColorToGray(roi, roi);
+                Cv2.GaussianBlur(roi, roi, new Size(3, 3), 0);
+                Cv2.MorphologyEx(roi, roi, MorphTypes.Close, ImageUtils.GetKernel(new Size(3, 3)));
+                Cv2.Canny(roi, roi, 50, 150);
+                Cv2.FindContours(roi, out Point[][] contours, out HierarchyIndex[] hierarchy, RetrievalModes.Tree, ContourApproximationModes.ApproxSimple);
+                foreach (Point[] point in contours)
```

Comment:
I think if you extracted this for each loop into a method you will pass guidelines for Better Code Hub

Reply:
I'm unsure why it failed the last one commit when the one before it passed.

Reply:
Working on refactoring